modules/renter/proto/negotiate.go

```diff
@@ -105,7 +104,7 @@ func verifyRecentRevision(conn net.Conn, contract modules.RenterContract) error
 	// check that revision number matches; if it does, do a more thorough
 	// check by comparing unlock hashes.
 	if lastRevision.NewRevisionNumber != contract.LastRevision.NewRevisionNumber {
-		return fmt.Errorf("our revision number (%v) does not match the host's (%v)", contract.LastRevision.NewRevisionNumber, lastRevision.NewRevisionNumber)
+		return &recentRevisionError{contract.LastRevision.NewRevisionNumber, lastRevision.NewRevisionNumber}
```

Comment:
The unlock hashes and signatures should be checked before the revision number, that way if a recent revision error is returned the caller can know that the only problem is the recent revision, doesn't have to worry about anything else.

Also, I might be missing something, but shouldn't the recent revision error also be returning the host signature? For the renter to be consistent, it should be saving the signatures.